executable-spec/test/Cardano/Ledger/Spec/STS/Chain/Chain/Properties.hs

```diff
@@ -441,14 +479,34 @@ lastStateContainsTallyOutcome
   -> Bool
 lastStateContainsTallyOutcome tr outc =
   let lastSt = Trace.lastState tr
-      vresips = Chain.vresips lastSt
+      vresips@(SIPsVoteResults (vresmap)) = Chain.vresips lastSt
       env = Trace._traceEnv tr
       sDist = Chain.stakeDist env
       pNoQ = Chain.prvNoQuorum env
       pNoM = Chain.prvNoMajority env
       r_a = Chain.r_a env
-  in any (\(_, outcome) -> outcome == outc )
-     $ Map.toList $ tallyOutcomeMap vresips sDist pNoQ pNoM r_a
+  in if outc /= Data.NoQuorum && outc /= Data.NoMajority
```

Comment:
We should use `case` statements to simplify these "tornado" :tornado: `if`s: it's really hard to understand what's going on and reason about the correctness of these functions.

Reply:
Done. OK